### ISSUE TITLE
[open directory] Open directory crashes

### ISSUE THREAD

**Issue by frosencrantz** [saulpw/visidata#1798]
**Small description**
Open directory crashes

**Expected result**
Doesn't crash

**Actual result with screenshot**
VisiData fails to start, and crashes:
```
AttributeError: module 'urllib' has no attribute 'error'
    visidata.main.vd_cli()
  File "/usr/lib/python3.9/site-packages/visidata/main.py", line 372, in vd_cli
    rc = main_vd()
  File "/usr/lib/python3.9/site-packages/visidata/main.py", line 294, in main_vd
    vs = vd.openSource(p, create=True, **opts) or vd.fail(f'could not open {p}')
  File "/usr/lib/python3.9/site-packages/visidata/_open.py", line 143, in openSource
    vs = vd.openPath(Path(p), filetype=filetype, create=create)  # convert to Path and recurse
  File "/usr/lib/python3.9/site-packages/visidata/_open.py", line 108, in openPath
    vs = openfunc(p)
TypeError: 'NoneType' object is not callable

```

**Steps to reproduce with sample data and a .vd**
`vd sample_data`

**Additional context**
Please include the version of VisiData. Latest development version.

I am wondering why the tests are passing.

**Comment by anjakefala** [comment]
Oh, I have been running into that! I thought that was because I was fiddling with the `git` filetype, and not for all directory openings.

**Comment by saulpw** [comment]
Should be fixed with 5251c3f1c33db8cba075f6efa2de75e92d696752

**Comment by anjakefala** [comment]
It does!

**Comment by frosencrantz** [comment]
Thank you @saulpw it is working now. 

 I think there might be an issue with the tests if they couldn't catch this bug for opening a directory.  Something to think about.

**Comment by anjakefala** [comment]
I just added one. =) I hope it works on the CI.

**Comment by saulpw** [comment]
There are 10,000 things to test and only 1,000 tests.  I know it seems like the answer is "add more tests", and I'm not opposed to that, but I had an addiction to tests ~20 years ago which had a severe negative impact on my coding life, and since then I've had to be more careful about how much energy I put into testing in general.  As always, if you're frustrated with the state of the `develop` branch, I would encourage you to stick with a released version, which will be more reliable.

**Comment by frosencrantz** [comment]
Thanks, @anjakefala, for adding the new tests, and helping make sure VisiData just continues to work well.

Sorry @saulpw  if I appear overzealous about the testing. I  try to be aware of your feelings about testing, I feel I overstepped this time.

 I expected opening a directory would be such basic functionality, if it was broken, it  would be caught by the tests.  I did check the tests and they were all green 😄, I looked at one of the test durations and it seemed very short,  which made me think the tests themselves were broken, too.   When I looked later I realized I don’t know how to read the test timing or don’t know how long the tests actually take.  (Now I’m wondering if Github provides some json feed or some other data source to see these tests results.)

And to be honest, I’ve been happy with all the development happening and how well VisiData code health has been.  I have braced myself for the development, and have setup multiple installations, so when there is a big breakage like this one, it doesn’t cause me issues.  Though I do like to be able to report bugs. I still need to provide a reproduction for https://github.com/saulpw/visidata/discussions/1732, which is still broken, but I have been very busy recently.
